framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -743,10 +746,30 @@ private void annotate(AnnotatedTypeMirror atype, Type typeDef) {
             WildcardType wildcardDef = (WildcardType) typeDef;
             if (wildcardDef.getExtends() != null) {
                 annotate(wildcardType.getExtendsBound(), wildcardDef.getExtends());
+                annotate(wildcardType.getSuperBound(), typeDef.getAnnotations());
             } else if (wildcardDef.getSuper() != null) {
                 annotate(wildcardType.getSuperBound(), wildcardDef.getSuper());
+                annotate(wildcardType.getExtendsBound(), typeDef.getAnnotations());
+            } else {
+                annotate(atype, typeDef.getAnnotations());
+            }
+        } else if (atype.getKind() == TypeKind.TYPEVAR) {
+            //Add annotations from the declaration of the TypeVariable
+            AnnotatedTypeVariable typeVarUse = (AnnotatedTypeVariable) atype;
+            for (AnnotatedTypeMirror annotatedTypeMirror : typeParameters) {
+                if (annotatedTypeMirror.getKind() != TypeKind.TYPEVAR) {
```

Comment:
Why do we need to allow this case? Can't we make the type of `typeParameters` more specific and only allow AnnotatedTypeVariables?